src/components/organisms/AssetActions/Compute/index.tsx

```diff
@@ -169,21 +169,30 @@ export default function Compute({
     ) {
       algorithmSelectionList = []
     } else {
-      const gueryResults = await queryMetadata(
+      const queryResults = await queryMetadata(
         getQuerryString(
           computeService.attributes.main.privacy.publisherTrustedAlgorithms,
           ddo.chainId
         ),
         source.token
       )
-      setDdoAlgorithmList(gueryResults.results)
-      const datasetComputeService = ddo.findServiceByType('compute')
-      algorithmSelectionList = await transformDDOToAssetSelection(
-        undefined,
-        gueryResults.results,
-        [],
-        newCancelToken()
+      // verify if the algorithms in the trusted list share the same endpoint of the dataset
+      const sameProviderEndpointAlgorithms = queryResults.results.filter(
+        (algo) =>
+          algo.service[1].serviceEndpoint === computeService.serviceEndpoint
```

Comment:
issue(blocking): There is a helper to get the correct service that we should use. Please have a look at `findServiceByType(type)` -> in this case I believe we are looking for the service of type "compute"

e.g.
```ts
(algo) => algo.findServiceByType('compute')?.serviceEndpoint === computeService.serviceEndpoint
```